src/components/product.js

```diff
@@ -65,7 +75,7 @@ export default class Product extends Component {
    * @param {Object} props - data and utilities passed down from UI instance.
    */
   constructor(config, props) {
-    super(config, props);
+    super(normalizeProductId(config), props);
```

Comment:
I'd use the same pattern we used in `js-buy-sdk`. It's kind of ugly to have a method that mutates its input. Can you receive the config then do `config.id = normalizeProductId(config)`?